From 95be250c13ec87ab59b4e20c071f68d616eb1d58 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <mselmeci@wisc.edu>
Date: Thu, 17 Apr 2025 10:42:11 -0500
Subject: [PATCH] Include additional XrdPfc headers

These are all the headers that are currently missing from cmake.
I'm not sure _all_ of these are needed but XrdPfcDirStateSnapshot.hh
is directly used by xrootd-lotman (and many of the rest are indirectly
needed).
---
 src/XrdPfc.cmake | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/src/XrdPfc.cmake b/src/XrdPfc.cmake
index 6d7107d3b..bbdcbddbd 100644
--- a/src/XrdPfc.cmake
+++ b/src/XrdPfc.cmake
@@ -38,7 +38,8 @@ add_library(
   XrdPfc/XrdPfcIO.cc            XrdPfc/XrdPfcIO.hh
   XrdPfc/XrdPfcIOFile.cc        XrdPfc/XrdPfcIOFile.hh
   XrdPfc/XrdPfcIOFileBlock.cc   XrdPfc/XrdPfcIOFileBlock.hh
-  XrdPfc/XrdPfcDecision.hh)
+  XrdPfc/XrdPfcDecision.hh
+  XrdPfc/XrdPfcTrace.hh)
 
 target_link_libraries(
   ${LIB_XRD_FILECACHE}
@@ -115,6 +116,17 @@ install(
     ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcFile.hh
     ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcTypes.hh
     ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcInfo.hh
+    ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcDecision.hh
+    ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcDirState.hh
+    ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcDirStateSnapshot.hh
+    ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcFPurgeState.hh
+    ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcFSctl.hh
+    ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcFsTraversal.hh
+    ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcIOFileBlock.hh
+    ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcIO.hh
+    ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcPrint.hh
+    ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcResourceMonitor.hh
+    ${CMAKE_CURRENT_SOURCE_DIR}/XrdPfc/XrdPfcTrace.hh
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/xrootd/XrdPfc
 )
 
-- 
2.43.5

